### PR TITLE
added support for transparend background (libwebkitgtk 2.7.4 min)

### PR DESCRIPTION
**Pull request by gchudnov**
* based on whether libwebkitgtk installed on the system has the version >= 2.7.4.

sample usage:
```
WebKit()
  .load(href)
  .wait('idle')
  .png('test.png'); // without transparent background (the same as before)
```
or
```
WebKit()
  .load(href)
  .wait('idle')
  .png('test.png', { transparent_background: true }); // use `true` | `false` to specify if the transparent background must be set in snapshot options.
```

can be checked for the following URL:
http://bob.provereno.ru/render/newbuildings/zverevo-derevnja-borisoglebskoe-161/badge.html

Let me know if there is something wrong or must be changed.
Or feel free to modify the code as needed.

--
FYI,
I built libwebkitgtk 2.7.4 the following way:
```
cmake .. -DPORT=GTK -DENABLE_WEB_AUDIO=OFF -DENABLE_VIDEO=OFF -DENABLE_VIDEO_TRACK=OFF
```

### CODE REVIEW

**Comment by gchudnov** [comment]
sorry, it looks I made a typo in a comment (2.7.4 of course)

**Comment by kapouer** [comment]
Thank you so much !
TLDR: load(url, {background: <css color, possibly rgba, defaults to white opaque>})
I think it'd be nicer to set transparent background by default on the snapshots and on the webview, and set document's background to a default white (to match current default behavior) - leaving the user the ability to change background color+alpha through a simple user stylesheet.

**Comment by kapouer** [comment]
I'll do that and release very very soon.

**Comment by gchudnov** [comment]
You're welcome.

Thank you for your efforts, node-webkitgtk is a great library!

**Comment by gchudnov** [comment]
hm... It looks there is an error in the code that I ussed as a pull request

I made in error with enum value.
it is not available for wegbitgtk that is less than 2.7.4

A conditional compilation is needed.
#ifdef 
#include "AAA"
#else
#include "BBB"
#endif
I can make a change and issue a new pull request if you wish

my fault, sorry

**Comment by gchudnov** [comment]
or there are some other options, without inclusion of separate files

**Comment by kapouer** [comment]
leave it like that, i'll rewrite it.

**Comment by kapouer** [comment]
I'm trying 
```
#include <webkit2/webkit2.h>
#ifdef WEBKIT_SNAPSHOT_OPTIONS_TRANSPARENT_BACKGROUND
#define default_snapshot_option WEBKIT_SNAPSHOT_OPTIONS_TRANSPARENT_BACKGROUND
#else
#define default_snapshot_option WEBKIT_SNAPSHOT_OPTIONS_NONE
#endif
```
but it doesn't see the first as defined (and it is - if i unconditionally compile i get transparent shots)

**Comment by gchudnov** [comment]
WEBKIT_SNAPSHOT_OPTIONS_TRANSPARENT_BACKGROUND is a value of [enum](https://github.com/WebKit/webkit/blob/7cba777081a871893d9f88354dc8bb7dcc010f26/Source/WebKit2/UIProcess/API/gtk/WebKitWebView.h#L168), not a macro-definition.

And WEBKIT_SNAPSHOT_OPTIONS_TRANSPARENT_BACKGROUND is not available in WebKitWebView.h for 2.6.x-versions

however, you can modify the code to 
```
#ifdef WEBKIT_CHECK_VERSION(2, 7, 4)
...
```
that should work

**Comment by kapouer** [comment]
indeed, it's `#if WEBKIT_CHECK_VERSION(2,7,4)` and it works, thank you.
This library will soon be able to draw html5 widgets without the window borders :)

**Comment by kapouer** [comment]
This is closed by https://github.com/kapouer/node-webkitgtk/releases/tag/1.13.0
